src/NuGet.Core/NuGet.Commands/TrustedSignersCommand/TrustedSignerActionsProvider.cs

```diff
@@ -0,0 +1,310 @@
+        public void AddOrUpdateTrustedSigner(string name, string fingerprint, HashAlgorithmName hashAlgorithm, bool allowUntrustedRoot)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, nameof(name));
+            }
+
+            if (string.IsNullOrEmpty(fingerprint))
+            {
+                throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, nameof(name));
+            }
+
+            if (!Enum.IsDefined(typeof(HashAlgorithmName), hashAlgorithm) || hashAlgorithm == HashAlgorithmName.Unknown)
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.UnsupportedHashAlgorithm, hashAlgorithm.ToString()));
+            }
+
+            var certificateToAdd = new CertificateItem(fingerprint, hashAlgorithm, allowUntrustedRoot);
+            TrustedSignerItem signerToAdd = null;
+
+            var signers = _trustedSignersProvider.GetTrustedSigners();
+            foreach (var existingSigner in signers)
+            {
+                if (string.Equals(existingSigner.Name, name, StringComparison.Ordinal))
+                {
+                    signerToAdd = existingSigner;
+
+                    break;
+                }
+            }
+
+            if (signerToAdd == null)
+            {
+                signerToAdd = new AuthorItem(name, certificateToAdd);
+            }
+            else
+            {
+                signerToAdd.Certificates.Add(certificateToAdd);
+            }
+
+            _trustedSignersProvider.AddOrUpdateTrustedSigner(signerToAdd);
+        }
+
+        /// <summary>
+        /// Adds a trusted repository with information from <paramref name="serviceIndex"/>
+        /// </summary>
+        /// <param name="name">Name of the trusted repository.</param>
+        /// <param name="serviceIndex">Service index of the trusted repository. Trusted certificates information will be gotten from here.</param>
+        /// <param name="owners">Owners to be trusted from the repository.</param>
+        /// <param name="token">Cancellation token</param>
+        public async Task AddTrustedRepositoryAsync(string name, Uri serviceIndex, IEnumerable<string> owners, CancellationToken token)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, nameof(name));
+            }
+
+            if (serviceIndex == null)
+            {
+                throw new ArgumentNullException(nameof(serviceIndex));
+            }
+
+            var signers = _trustedSignersProvider.GetTrustedSigners();
+            foreach (var existingSigner in signers)
+            {
+                if (string.Equals(existingSigner.Name, name, StringComparison.Ordinal))
```

Comment:
why aren't these case insensitive? 
Why shouldn't you use OrdinalIgnoreCase?

Reply:
This property is not case sensitive https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Settings/Items/AuthorItem.cs#L72